core/app/models/concerns/spree/single_store_resource.rb

```diff
@@ -3,7 +3,9 @@ module SingleStoreResource
     extend ActiveSupport::Concern
 
     included do
-      scope :for_store, ->(store) { where(store_id: store.id) }
+      if has_attribute?(:store_id)
```

Comment:
@MatthewKennedy this will not work, I've meant moving the validation for setting store_id to a different store if already one is set